src/hough.rs

```diff
@@ -81,7 +82,7 @@ pub fn detect_lines(image: &GrayImage, options: LineDetectionOptions) -> Vec<Pol
             let votes = unsafe { acc_sup.unsafe_get_pixel(r, m)[0] };
             if votes >= options.vote_threshold {
                 let line = PolarLine {
-                    r: r as f32,
+                    r: r as f32 - rmax as f32,
```

Comment:
For large images (diagonal longer than `1 << 23`), the cast and subtraction are inaccurate in `f32` but at least the subtraction would be accurate in `u32/i32`. Maybe rewrite as `(r - rmax as u32) as i32 as f32`. 

The inaccuracy affects all lines for larger images. Especially those detected with an offset close to `0` have a high relative error. After the change, only affectes lines with an offset greater than `1 << 23`.